### DISCUSSION TITLE
[Docker] Suggestion: support base path for selfhosting

### DISCUSSION THREAD

**Discussion by sambartik** [themepark-dev/theme.park#353] [General]
Hello, I think it would be really handy to support base path so that the css can be accessed behind reverse proxy with url: `xxx.xxx.xxx.xxx/themepark`.

**Comment by GilbN** [comment]
please test `-e TP_URLBASE` on the `develop` tag

**Comment by sambartik** [comment]
Thanks, great work! Seems to be okay. However it is not yet implemented in _docker mods_ for me and putting `xxx.xxx.xxx.xxx/themepark` url in `TP_DOMAIN` env variable resulted in sed errors.

**Comment by GilbN** [comment]
use `TP_DOMAIN=domain.com\/themepark`

**Comment by sambartik** [comment]
Great, it works! Thanks!

**Comment by sambartik** [comment]
I have just noticed that when accessing the base theme css files via web, they have an @import statements with url that has `http` scheme despite having `TP_SCHEME=https` env variable set up. When I `cat` the css file, those same urls had `https` scheme, but they were missing the base path.

So: when acessing 
- Via web: `@import url("http://xxx.xxx.xxx.xxx/themepark/css/base/bazarr/bazarr-base.css?sha=None");`
- Directly the css file located in `/config/www/css` folder: `@import url("https://xxx.xxx.xxx.xxx/css/base/bazarr/bazarr-base.css?sha=None");`

**Comment by GilbN** [comment]
make sure you reverse proxy the https port or access locally using the https port

**Comment by sambartik** [comment]
Yup, I was connecting to theme-park via http from the reverse proxy point of view. Nonetheless, it seems kind of strange. I would think that the scheme depended entirely on what I set in the env variable. Is this the intended behaviour?

**Reply by GilbN** [reply]
Yeah, even though your reverse proxy was using https it didn't pass its $scheme to the upstream server, and $scheme in that config was read as http since you proxied through the http port. 
https://github.com/GilbN/theme.park/blob/9b2d43722dc49df0526c4975350384abb81ad1a4/docker/root/defaults/default#L19-L22

I'm gonna add a new sed of `$scheme` and change the default value of `TP_SCHEME` to `$scheme` if it's not used. 

That way you dont really need to add any ENVs for local use as it will switch automatically between the two. 
And if you are reverse proxying with HTTPS you can either add `TP_SCHEME=https` or reverse proxy the upstream https port. 

Also if you add `proxy_set_header Host $host;` in your reverse proxy you dont even need to add `TP_DOMAIN` as it will pass the `$host` var from your webserver to the upstream and change the `$http_host` var which is what `TP_DOMAIN` defaults to.
